src/withStyles.jsx

```diff
@@ -51,31 +51,82 @@ export function withStyles(
+  }
+
+  /**
+   * Cache for storing the results of computations:
+   * `WeakMap<Theme, WeakMap<typeof WithStyles, { ltr: {}, rtl: {} }>>`
+   * Falling back to `Map` whenever `WeakMap` is not supported
+   */
+  const withStylesCache = typeof WeakMap === 'undefined' ? new Map() : new WeakMap();
+
+  function getComponentCache(theme, component, direction) {
+    const themeCache = withStylesCache.get(theme);
+    if (!themeCache) {
+      return null;
+    }
+    const componentCache = themeCache.get(component);
+    if (!componentCache) {
+      return null;
+    }
+    return componentCache[direction];
+  }
+
+  function updateComponentCache(theme, component, direction, results) {
+    let themeCache = withStylesCache.get(theme);
+    if (!themeCache) {
+      themeCache = typeof WeakMap === 'undefined' ? new Map() : new WeakMap();
+      withStylesCache.set(theme, themeCache);
+    }
+    let componentCache = themeCache.get(component);
+    if (!componentCache) {
+      componentCache = { ltr: {}, rtl: {} };
+      themeCache.set(component, componentCache);
+    }
+
+    componentCache[direction] = results;
+  }
+
+  /** Derive the create function from the interface and direction */
+  function makeCreateFn(direction, stylesInterface) {
+    const directionSelector = direction === DIRECTIONS.RTL ? 'RTL' : 'LTR';
+    let create = stylesInterface[`create${directionSelector}`] || stylesInterface.create;
+    if (process.env.NODE_ENV !== 'production') {
+      create = withPerf('create')(create);
+    }
+    return create;
+  }
+
+  /** Derive the resolve function from the interface and direction */
+  function makeResolveFn(direction, stylesInterface) {
+    const directionSelector = direction === DIRECTIONS.RTL ? 'RTL' : 'LTR';
+    let resolve = stylesInterface[`resolve${directionSelector}`] || stylesInterface.resolve;
+    if (process.env.NODE_ENV !== 'production') {
+      resolve = withPerf('resolve')(resolve);
+    }
+    return resolve;
+  }
+
   // The function that wraps the provided component in a wrapper
   // component that injects the withStyles props
   return function withStylesHOC(WrappedComponent) {
     const wrappedComponentName = getComponentName(WrappedComponent);
 
     // The wrapper component that injects the withStyles props
     class WithStyles extends BaseClass {
-      constructor(props, context) {
-        super(props, context);
-
-        /** Cache for storing the current props and objects used to derive them */
-        this.cache = { rtl: {}, ltr: {} };
-      }
-
-      getCache(direction) {
-        return this.cache[direction];
-      }
-
-      updateCache(direction, results) {
-        this.cache[direction] = results;
-      }
-
       getCurrentInterface() {
         // Fallback to the singleton implementation
-        return (this.context && this.context.stylesInterface) || _getInterface();
+        return (
+          (this.context && this.context.stylesInterface) || _getInterface()
```

Comment:
nbd ofc but this change seems surprising. is your editor running more than eslint autofix on things?

Reply:
@ljharb You're right this is weird, lol. I have some weird linting/prettier/editor autofix stuff going on 🤦‍♀ I tried fixing it but it's still giving me things like this.

Reply:
Generally the only formatting you want enabled is eslint autofix - ie, in vscode, disable "format on save", and prettier, and be sure you don't have any global eslint installed (so it always uses the local one)